testing/camunda-process-test-java/src/main/java/io/camunda/process/test/impl/runtime/ContainerRuntimePropertiesUtil.java

```diff
@@ -0,0 +1,190 @@
+  private final String camundaDockerImageVersion;
+  private final String connectorsDockerImageName;
+  private final String connectorsDockerImageVersion;
+  private final String elasticsearchVersion;
+
+  public ContainerRuntimePropertiesUtil(final Properties properties) {
+    camundaVersion =
+        getLatestReleasedVersion(
+            properties,
+            PROPERTY_NAME_CAMUNDA_VERSION,
+            ContainerRuntimeDefaults.DEFAULT_CAMUNDA_DOCKER_IMAGE_VERSION);
+    elasticsearchVersion =
+        getPropertyOrDefault(
+            properties,
+            PROPERTY_NAME_ELASTICSEARCH_VERSION,
+            ContainerRuntimeDefaults.DEFAULT_ELASTICSEARCH_VERSION);
+
+    camundaDockerImageName =
+        getPropertyOrDefault(
+            properties,
+            PROPERTY_NAME_CAMUNDA_DOCKER_IMAGE_NAME,
+            ContainerRuntimeDefaults.DEFAULT_CAMUNDA_DOCKER_IMAGE_NAME);
+    camundaDockerImageVersion =
+        getLatestReleasedVersion(
+            properties,
+            PROPERTY_NAME_CAMUNDA_DOCKER_IMAGE_VERSION,
+            ContainerRuntimeDefaults.DEFAULT_CAMUNDA_DOCKER_IMAGE_VERSION);
+
+    connectorsDockerImageName =
+        getPropertyOrDefault(
+            properties,
+            PROPERTY_NAME_CONNECTORS_DOCKER_IMAGE_NAME,
+            ContainerRuntimeDefaults.DEFAULT_CONNECTORS_DOCKER_IMAGE_NAME);
+    connectorsDockerImageVersion =
+        getLatestReleasedVersion(
+            properties,
+            PROPERTY_NAME_CONNECTORS_DOCKER_IMAGE_VERSION,
+            ContainerRuntimeDefaults.DEFAULT_CONNECTORS_DOCKER_IMAGE_VERSION);
+  }
+
+  private static String getLatestReleasedVersion(
+      final Properties properties, final String propertyName, final String defaultValue) {
+
+    final String propertyValue = properties.getProperty(propertyName);
+    if (propertyValue == null || isPlaceholder(propertyValue)) {
+      return defaultValue;
+    }
+
+    return Optional.of(propertyValue)
+        .map(SEMANTIC_VERSION_PATTERN::matcher)
+        .filter(Matcher::find)
+        .map(
+            matcher -> {
+              final int major = Integer.parseInt(matcher.group(1));
+              final int minor = Integer.parseInt(matcher.group(2));
+              final int patch = Integer.parseInt(matcher.group(3));
+              final String label = matcher.group(4);
+              return getLatestReleasedVersion(major, minor, patch, label);
+            })
+        .orElse(propertyValue);
+  }
+
+  private static boolean isPlaceholder(final String propertyValue) {
+    return PLACEHOLDER_PATTERN.matcher(propertyValue).matches();
+  }
+
+  private static String getLatestReleasedVersion(
+      final int major, final int minor, final int patch, final String label) {
+
+    if (label == null) {
+      // release version
+      return String.format(VERSION_FORMAT, major, minor, patch);
+
+    } else if (patch == 0) {
+      // current dev version
+      return SNAPSHOT_VERSION;
```

Comment:
💭 The `SNAPSHOT_VERSION` is probably technically not the latest released version, so the method name does not reflect that. But also I have a hard time thinking of a better name for the method.

Reply:
You're right. Thank you for sharing your thoughts. We will try to find a better name when we tough the method again. :sweat_smile: